common/engine/Stack.lua

```diff
@@ -2011,11 +2011,11 @@ function Stack.onGarbageLand(self, panel)
       -- to prevent from running this code dozens of time for the same garbage block
       -- all panels of a garbage block have the same id + shake time
       self.garbageLandedThisFrame[#self.garbageLandedThisFrame+1] = panel.garbageId
+      self:emitSignal("garbageLanded", self, panel.shake_time)
     end
 
     -- whether we ran through it or not, the panel should lose its shake time
     panel.shake_time = nil
-    self:emitSignal("hurt", "hurt")
   end
```

Comment:
I moved this up here because putting it below only causes multiple redundant signals on the same frame.
Panel physics work on a per-panel basis even with garbage that functionally has multiple panels working as one.
So if you have a 6 wide piece of garbage, you'd send the signal for every single panel in that row even though they all have the same shake time value. There is a redundancy check a bit above that makes sure we only get shake time reported once per piece of garbage and including the signal reporting here makes sense.